src/mesh/structured_mesh.jl

```diff
@@ -0,0 +1,15 @@
+struct StructuredMesh{RealT<:Real, NDIMS}
+  size::Tuple{Vararg{Int64, NDIMS}}
+  coordinates_min::Tuple{Vararg{RealT, NDIMS}}
+  coordinates_max::Tuple{Vararg{RealT, NDIMS}}
+  linear_indices::LinearIndices{NDIMS, Tuple{Vararg{UnitRange{Int64}, NDIMS}}}
+end
+
+function StructuredMesh{RealT}(size, coordinates_min, coordinates_max) where {RealT<:Real}
+  NDIMS = length(size)
+
+  return StructuredMesh{RealT, NDIMS}(size, coordinates_min, coordinates_max, LinearIndices(size))
+end
```

Comment:
I think it would be nice to drop the parameter `RealT` and replace it by something like
```julia
RealT = promote_type(eltype(coordinates_min), coordinates_max))
```
performing appropriate conversations afterwards.

Reply:
Shouldn't it rather be
```julia
RealT = promote_type(eltype(coordinates_min), eltype(coordinates_max))
```
?

Reply:
Whoops, sure - looks like I was too tired yesterday :sweat_smile:

Reply:
However, I am not sure about using `promote_type` in general. In [`SerialTree`](https://github.com/trixi-framework/Trixi.jl/blob/main/src/mesh/serial_tree.jl) for instance, we use hardcoded `Float64` for the coordinates and other physical size-related quantities. Otherwise, specifying coordinates_min/max as `[-2, -2]` and `[2, 2]` would result in integer coordinates, which is not what we want. Maybe instead pass `RealT=Float64` as a keyword argument such that it uses a sensible default but can be overruled?

Reply:
Well, I prefer educating users to specify the correct types of coordinates they want to use (just append `.0` to their integer coordinates if necessary). If we hard-code `Float64`, we can't compute derivatives with respect to mesh topology etc. and cripple the opportunities for differentiable programming on purpose.

Reply:
No, I fully understand and support your reasoning. I just thought it might be consistent to what we have done in the tree mesh part of Trixi. But different solver, different mesh, different rules - I could thus very well live with what you suggested as well :+1:

Reply:
Well, my suggestion would include moving to general types for the `TreeMesh` as well in the future

Reply:
Done